rio_tiler/io/rasterio.py

```diff
@@ -545,6 +545,20 @@ def feature(
             **kwargs,
         )
 
+        cutline_mask = rasterize(
+            [shape],
```

Comment:
@yellowcap I think there is something we are missing here. 

shape is assumed to be in the same CRS as the `img` crs but this is only the case when the user pass `dst_crs` equal `shape_crs` or `dst_crs=None` (because we always default to `shape_crs`.

before creating the `cutline_mask` I think you'll need to reproject the shape to the `img.crs`

we should also had better test because I feel this should have been caught (by older test)

Reply:
```python
# Using New feature method 
with Reader(src_path) as f:
    img_f = f.feature(feat, dst_crs="epsg:3857")
    
# Using GDAL cutline 
with Reader(src_path) as f:
    cutline = create_cutline(f.dataset, feat, geometry_crs="epsg:4326")
    bbox = featureBounds(feat)
    img = f.part(bbox, vrt_options={"cutline": cutline}, dst_crs="epsg:3857")  

numpy.testing.assert_array_equal(img.mask, img_f.mask)
```

taken from https://github.com/cogeotiff/rio-tiler/pull/598/files#r1182322351

we can confirm this `bug` with ☝️

Reply:
no worries 😄